### PR TITLE
Add InitConfig.ConserveMemory and set EnvironmentVariable based on it

### PR DESCRIPTION
**Pull request by LukaszRozmej**
## Changes

- Add InitConfig.ConserveMemory and set EnvironmentVariable based on it.
- Add CommandRunner to run custom OS commands to set EnvironmentVariable.

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [x] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

### CODE REVIEW

File: src/Nethermind/Nethermind.Runner/EnvironmentVariables/CommandRunner.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        }
+
+        return (-1, $"Couldn't start process {info.FileName} {info.Arguments}", string.Empty);
+    }
+
+    private static string GetArguments(string command) =>
+        RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? $"/C {command}" : $"-c \"{command}\"";
+
+    private static string GetFileName() =>
+        RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "cmd.exe" : "bash";
```
Comment (flcl42) [inline src/Nethermind/Nethermind.Runner/EnvironmentVariables/CommandRunner.cs:35]: Bash is not everywhere, I'd test [useshellexecute](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.processstartinfo.useshellexecute?view=net-7.0)

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Runner/EnvironmentVariables/CommandRunner.cs]: Do you know it works? Seems it only allows for `open`: https://github.com/tmds/corefx/blob/master/src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs#L405-L412

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Runner/EnvironmentVariables/CommandRunner.cs:35]: I tried and couldn't make it working.

**Comment by LukaszRozmej** [comment]
Couldn't make it work on linux

**Comment by LukaszRozmej** [comment]
Didn't work

**Pull request closed without merging**